crates/circuit/src/dag_circuit.rs

```diff
@@ -521,12 +519,49 @@ impl DAGCircuit {
                 )
             ),
             py.get_type::<PyDeprecationWarning>(),
-            2,
+            1,
         ))?;
         Ok(self.duration.as_ref().map(|x| x.clone_ref(py)))
```

Comment:
We probably could have done this to deduplicate the code a little bit:

```suggestion
        self.get_internal_duration(py)
```

but it's not a big deal. Not worth blocking over.

Reply:
I had to change a small detail to fix lint so I can also fix this